src/doc.d

```diff
@@ -2371,6 +2497,47 @@ extern (C++) void highlightCode(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
                 i = j - 1;
             }
         }
+        else if (!resolvedTemplateParameters)
+        {
+            // hunt for template declarations:
+            for (size_t symi = 0; symi < a.dim; symi++)
+            {
+                FuncDeclaration fd = (*a)[symi].isFuncDeclaration();
+
+                if (!fd || !fd.parent.isTemplateDeclaration())
+                {
+                    continue;
+                }
+
+                TemplateDeclaration td = fd.parent.isTemplateDeclaration();
+
+                // build the template parameters
+                OutBuffer parametersBuf;
```

Comment:
Can I get feeback on lines 2515-2517 specifically? I copied this almost verbatim out of https://github.com/D-Programming-Language/dmd/blob/d9e91cc911c8bf4a81bee929fa8768c02a1b3a3d/src/dtemplate.d#L645-L656. Only difference is that we don't care about the ident. This could easily be refactored into one method contained on the `TemplateDeclaration`... something like `TemplateDeclaration::writeParameters(OutBuffer*)` or similar so that consistency is guaranteed.